src/irmin-pack/traverse_pack_file.ml

```diff
@@ -56,7 +61,65 @@ end = struct
      as requests for more data. *)
   exception Not_enough_buffer
 
-  type index_binding = { key : Hash.t; data : Index.value }
+  type index_value = int63 * int * Pack_value.Kind.t
+  [@@deriving irmin ~equal ~pp]
+
+  type index_binding = { key : Hash.t; data : index_value }
+
+  module Index_reconstructor = struct
+    let create ?output config =
+      let dest =
+        match output with
+        | Some path -> path
+        | None ->
+            if Conf.readonly config then raise S.RO_not_allowed;
```

Comment:
I moved this check to the `output = None` branch

Reply:
But maybe I should move it out, because `path` in `Some path` could be `root`, and I'm only aware of [Core's realpath function](https://ocaml.janestreet.com/ocaml-core/109.20.00/doc/core/Filename.html#VALrealpath) as a solution to check the equality of 2 directories

Reply:
Ah, I just commented something similar. It might make sense the `path` if it already exists as a file at all (regardless of whether or not that file is the original index or not).